libcanard/canard.c

```diff
@@ -1104,38 +1107,35 @@ int32_t canardTxPush(CanardTxQueue* const                que,
     return out;
 }
 
-CanardTxQueueItem* canardTxPeek(const CanardTxQueue* const que)
+struct CanardTxQueueItem* canardTxPeek(const struct CanardTxQueue* const que)
 {
-    CanardTxQueueItem* out = NULL;
+    struct CanardTxQueueItem* out = NULL;
     if (que != NULL)
     {
         // Paragraph 6.7.2.1.15 of the C standard says:
         //     A pointer to a structure object, suitably converted, points to its initial member, and vice versa.
-        out = (CanardTxQueueItem*) (void*) cavlFindExtremum(que->root, false);
+        out = (struct CanardTxQueueItem*) (void*) cavlFindExtremum(que->root, false);
     }
     return out;
 }
 
-CanardTxQueueItem* canardTxPop(CanardTxQueue* const que, const CanardTxQueueItem* const item)
+struct CanardTxQueueItem* canardTxPop(struct CanardTxQueue* const que, struct CanardTxQueueItem* const item)
 {
-    CanardTxQueueItem* out = NULL;
     if ((que != NULL) && (item != NULL))
     {
-        // Intentional violation of MISRA: casting away const qualifier. This is considered safe because the API
-        // contract dictates that the pointer shall point to a mutable entity in RAM previously allocated by the
-        // memory manager. It is difficult to avoid this cast in this context.
-        out = (CanardTxQueueItem*) item;  // NOSONAR casting away const qualifier.
```

Comment:
As far as we now have "mutable" item from `canardTxPeek`, I decided also change signature of `canardTxPop` by removing `const` from the `item` parameter. Hence no need anymore to violate MISRA. Gonna do similar change at Udpard as well.